upa/contracts/UpaProofReceiver.sol

```diff
@@ -407,7 +402,6 @@ contract UpaProofReceiver is
 
         uint256[] memory publicInput = publicInputs[i];
         bytes32 proofId = UpaLib.computeProofId(circuitId, publicInput);
-        require(proofId != DUMMY_PROOF_ID, DummyProofInSubmission());
```

Comment:
I definitely get the point.  On the other hand I like the idea of completely eliminating this whole branch of possible states of the protocol.  Unless you feel strongly I'm inclined to keep this.  @SupremoUGH @dysfunctor what do you think?

Reply:
Yeah, I'm also a bit inclined to keep it for the same reason.